### ISSUE TITLE
Accessibility errors related to missing alt texts and low contrast.

### ISSUE THREAD

**Issue by oldrup** [wolfgang101/include-mastodon-feed#14]
Hey!

Quick test on WebAIM wave. I expect this to be reasonable easy fixes, given that we already have an image title text. Will get back with some recommendations once I've done some more testing including synthetic tests and tests with screenreaders and keyboard navigation 🌞

https://wave.webaim.org/report#/https://test.oldrup.dev/mastodon/

![Image](https://github.com/user-attachments/assets/6b54e814-17fa-4a49-b296-4639588aa0aa)

Bjarne

**Comment by wolfgang101** [comment]
I already published version 1.12 which adds the alt attribute to images. That seemed like the most severe offense.

Not sure yet what the best way to go about the contrast issue is though:

* it concerns primarily the default link color
* it's the Mastodon brand color and even on the native Mastodon interface they use the same font / background color combination
* the color in general can be easily customized by setting a PHP constant or a custom style
* might be a fix to make the affected strings bold instead of changing color?

Looking forward to hear about your suggestions.

**Comment by oldrup** [comment]
Awesome, Wolfgang!. There are definetely low-hanging fruits that can improve the user experience for many people. I'm so glad you're giving this some thoughts and time.

Quick feedback, so far:

- I wasn't able to get the alt text (image descriptions) to show. Will do more troubleshooting and report back.
- I'm testing accessibility on a simpler site https://test.oldrup.dev/mastodon/accessibility/. It's in active development.
- Regarding the contrasts, the pantone 2725 on https://joinmastodon.org/branding works fine by me on a white background, that is. I'm using the built in variables of the plugin to set it, just thinking "accessible defaults" might be a good idea for those who aren't aware.

What I'm looking further into:

- Tabbing order and screen reader text (css ORDER looks fine visually, but messes up TABINDEX)
- Description of external links? (cue that links leads to mastodon?)
- HTML Semantics of the list of "cards"? (DIV vs OL/LI)
- Duplicate links (image link and meta data date)

Once I have a reasonably solid suggestion, I'll let you know :)

Should we contribute a couple of hours to https://equalizedigital.com/gaad2025/ ?

**Comment by wolfgang101** [comment]
Awesome idea! I just submitted a 3 hr pledge. Not sure if it gets published though, as I did not share my full name.

**Comment by oldrup** [comment]
Fantastic! I pledged 3 hours as well. That definitely goes a long way, given your insane momentum 😄 I'll see if I can prepare qualified recommendations so we can dive into actual decision-making, implementation, testing, and documentation on the day.

Thank you!
